sled-agent/src/instance.rs

```diff
@@ -1519,3 +1519,634 @@ impl InstanceRunner {
         out
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
As discussed in chat - this should be illumos-only if it's using the `StorageManagerTestHarness`

Reply:
Done in [b7c1d67](https://github.com/oxidecomputer/omicron/pull/5294/commits/b7c1d67c7c215d6e231a2728e4d7b049e5706637)